nixpkgs/nixpkgs.bzl

```diff
@@ -25,19 +25,27 @@ def _nixpkgs_package_impl(ctx):
   else:
     ctx.template("BUILD", Label("@io_tweag_rules_nixpkgs//nixpkgs:BUILD.pkg"))
 
-  path = '<nixpkgs>'
+  # If neither repository or path are set, leave empty which will use
+  # default value from NIX_PATH
+  path = ''
   if ctx.attr.repository and ctx.attr.path:
     fail("'repository' and 'path' fields are mutually exclusive.")
   if ctx.attr.repository:
     # XXX Another hack: the repository label typically resolves to
     # some top-level package in the external workspace. So we use
     # dirname to get the actual workspace path.
-    path = ctx.path(ctx.attr.repository).dirname
+    path = "-I nixpkgs={0}".format(ctx.path(ctx.attr.repository).dirname)
   if ctx.attr.path:
-    path = ctx.attr.path
+    path = "-I nixpkgs={0}".format(ctx.attr.path)
 
   attr_path = ctx.attr.attribute_path or ctx.name
-  res = ctx.execute(["nix-build", path, "-A", attr_path, "--no-out-link"])
+  # Use -E instead of -A: this way we can use build arbitrary
+  # attributes rather than being limited to what -A accepts. An
+  # example of this is haskellPackages.ghcWithPackages which is
+  # unusable with -A.
+  res = ctx.execute(["nix-build", path, "-E",
+                     "(import <nixpkgs> {{}}).{0}".format(attr_path),
```

Comment:
Also, if an attribute path is now really an expression, then the name should be changed and it should be documented.